cql3/restrictions/statement_restrictions.cc

```diff
@@ -1021,5 +1022,28 @@ ::shared_ptr<single_column_restriction> single_column_restriction::LIKE::apply_t
     return r;
 }
 
+namespace wip {
+
+namespace {
+
+using children_t = std::vector<expression>; // conjunction's children.
+
+children_t explode_conjunction(const expression& e) {
+    return std::visit(overloaded_functor{
+            [] (const conjunction& c) { return c.children; },
+            [&] (const auto&) { return children_t{e}; },
+        }, e);
+}
+
+} // anonymous namespace
+
+expression make_conjunction(const expression& a, const expression& b) {
```

Comment:
We should take the parameters by value so if the caller moves a value in, we elide the copy. But I'm fine with delaying until later.

Reply:
I'm not sure how that would elide the copy on line 1033, which is the only actual waste (and only if the result of make_conjunction is assigned right back to the argument `a`).

Reply:
explode_conjunction() would also accept by value, and return std::move(e.children).

Reply:
Done as suggested.